src/applications/mhv-medical-records/actions/allergies.js

```diff
@@ -1,28 +1,57 @@
+import { selectIsCernerPatient } from '~/platform/user/cerner-dsot/selectors';
 import { Actions } from '../util/actionTypes';
 import {
   getAllergies,
   getAllergy,
   getAcceleratedAllergies,
   getAcceleratedAllergy,
+  getAllergiesWithOHData,
 } from '../api/MrApi';
 import * as Constants from '../util/constants';
 import { addAlert } from './alerts';
 import { dispatchDetails } from '../util/helpers';
 import { getListWithRetry } from './common';
 
-export const getAllergiesList = (
-  isCurrent = false,
-  isAccelerating = false,
-) => async dispatch => {
+export const getAllergiesList = (isCurrent = false) => async (
+  dispatch,
+  getState,
+) => {
   dispatch({
     type: Actions.Allergies.UPDATE_LIST_STATE,
     payload: Constants.loadStates.FETCHING,
   });
   try {
-    const getData = isAccelerating ? getAcceleratedAllergies : getAllergies;
+    const state = getState();
+    const isCerner = selectIsCernerPatient(state);
+
+    // Determine acceleration from state
+    const isAcceleratedDeliveryEnabled =
+      state.featureToggles?.mhvAcceleratedDeliveryEnabled;
+    const isAcceleratingAllergiesEnabled =
+      state.featureToggles?.mhvAcceleratedDeliveryAllergiesEnabled;
+    const isAccelerating =
```

Comment:
BLOCKER: I would not duplicate this logic in the reducer and use the logic in the hook and pass in the values via parameters